core/src/main/java/io/questdb/cutlass/text/AbstractTextLexer.java

```diff
@@ -326,15 +304,25 @@ private void parse0(long lo, long hi) {
                     if (zeroBytesWord == 0) {
                         ptr += 7;
                         this.fieldHi += 7;
+                        this.ascii &= (word & NON_ASCII_MASK_FULL) == 0;
                         continue;
                     } else {
                         int firstIndex = SwarUtils.indexOfFirstMarkedByte(zeroBytesWord);
-                        ptr += firstIndex;
+                        if (firstIndex > 0) {
+                            // The firstIndex returns the byte count we have to "trust"
+                            // the assumption is that these bytes will become a part of the existing fields
+                            // These bytes come on LOW bits of the "word". To check that these bytes are
+                            // positive, we need to isolate them. We do that by masking out the entire
+                            // word, save for the bytes we intend to keep.
+                            this.ascii &= ((word & (0xffffffffffffffffL >>> (64 - firstIndex * 8))) & NON_ASCII_MASK_FULL) == 0;
+                            ptr += firstIndex;
+                        }
                         this.fieldHi += firstIndex;
                     }
                 }
 
                 final byte b = Unsafe.getUnsafe().getByte(ptr++);
+                this.ascii = this.ascii && b > 0;
```

Comment:
```suggestion
                this.ascii &= b > 0;
```

Reply:
fixed in https://github.com/questdb/questdb/pull/4401/commits/765e27891d2394b5680b41a1483611d61cbb6cc6